plugins/woocommerce-blocks/assets/js/blocks/mini-cart/edit.tsx

```diff
@@ -87,7 +91,58 @@ const Edit = ( { attributes, setAttributes }: Props ): ReactElement => {
 		''
 	) as string;
 
-	const productCount = 0;
+	/**
+	 * This is a workaround for the Site Editor to set the correct
+	 * background color of the Mini-Cart QuantityBadge block based on
+	 * the main background color set by the theme.
+	 */
```

Comment:
Hi @tarunvijwani,

These [two CSS properties](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/quantity-badge/style.scss#L28-L29) are hiding the count badge in the Mini Cart block. So earlier I changed these values to hard-coded colors which isn't a good approach since the goal is to keep product count badge background to be the theme's foreground color, and it's text color should be the theme's background color. So for this reason I have used this workaround recommended by @/opr (which is also used in this [file](https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce-blocks/assets/js/blocks/mini-cart/mini-cart-contents/edit.tsx#L62-L113))

https://github.com/woocommerce/woocommerce/assets/77401999/ddc1d05f-95b3-475b-b53c-c25acd2599e0

But I see one problem, is that it works in the post editor but it doesn't work in the site editor since the content comes under iframe (which isn't the case in post editor) so I have to refine this `useEffect`

> Could you please help me with before and after images for this one along with the steps to reproduce it?

To test this:
1. Go the Side editor > Global Styles
2. Go to the `Colors` settings from the inspector control
3. Try changing the background color and text color and see the opposite will get applied to the product count badge to keep the styles consistent.
4. This will work in any posts where Mini Cart block is added. (For it to be work on site editor, I'll have to change in useEffect)